include/imgproc/dct_hash.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// Standard C++ includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <bitset>
+// Standard C includes
+#include <cmath>
+
+//opencv
+#include <opencv2/opencv.hpp>
+
+namespace DCTHash
+{
+    //! computes the DCT hash
+    inline static std::bitset<64> computeHash(const cv::Mat &in) {
+
+        cv::Mat dct_mat;
+        cv::dct(in, dct_mat);
+        cv::Mat rect(dct_mat,{ 0, 0, 8, 8});                // we only need the low 8x8 frequencies
+        std::sort(rect.begin<float>(),rect.end<float>());   // sorting to get median value
+        const float median= rect.at<float>(-1+rect.size().height/2,rect.size().width-1); 
+        rect = {dct_mat,{ 0, 0, 8, 8}};                     // re-assigning so we get the correct order back
```

Comment:
I've only just noticed this, but this is a major bug! You *won't* get the correct order back if you do this. When you change ``rect`` you also change ``dct_mat``: OpenCV does not make a copy of the matrix.

Reply:
Instead of sorting ``rect`` in place, I'd suggest this:
```c++
std::array<float, 33> sorted;
std::partial_sort_copy(rect.begin<float>(), rect.end<float>(), sorted.begin(), sorted.end());
```

``sorted`` is declared on the stack so you don't have unnecessary heap allocations.